### ISSUE TITLE
If retry happens, no matter the test is success or not, the Jenkins job fails 

### ISSUE THREAD

**Issue by Kenith** [katalon-studio/katalon-studio#42]
Hello All,

These days we are using docker to run the test in Jenkins.
We found that If retry happens, no matter the test is success or not, the jenkins job fails.

Please take time to fix it.

Thank you,
Ken

**Comment by devalex88** [comment]
I've moved this issue to https://github.com/katalon-studio/katalon-studio for planning.

**Comment by donwelsh** [comment]
I am facing this issue as well. Any plan to fix soon?

**Comment by amyzhuyx** [comment]
I am also facing this issue. When will it by fixed?

**Comment by jeff-hamblen** [comment]
I would just like to bump this thread for an ETA. I am experiencing the same problem.

**Comment by donwelsh** [comment]
Any chance this will be fixed soon? It is causing me a lot of problems at work.

**Comment by devalex88** [comment]
We are working on it for release 6.1.3.

**Comment by donwelsh** [comment]
Thanks for the status on this @devalex88

**Comment by donwelsh** [comment]
@devalex88 I see this didn't make it in 6.1.3, 6.1.4 or the latest 6.1.5. Is it still being worked on?

Thanks

**Comment by jeff-hamblen** [comment]
Any update on this? It does not appear to be in the latest release (6.1.5). It also does not appear to be in the 6.1.6 beta build notes seen here https://forum.katalon.com/t/katalon-studio-6-1-6-beta/25892

**Comment by jonwest3** [comment]
I am running into this problem as well. Any update on when the fix will be available?

Thank you

**Comment by devalex88** [comment]
Please try the latest version of Katalon Studio. The issue has been fixed already. If it still happens on your side, please provide some description and we will resolve it in 6.4.0.

**Comment by jhowellbm** [comment]
I can confirm this is still a bug in Katalon 6.3.3.

**Comment by donwelsh** [comment]
I also can confirm this is still a bug in Katalon 6.3.3. For me I am also running a test suite collection on Jenkins. One of my test fails, is retried and passes successfully, but the Jenkins job is still marked as a failure.

**Comment by Kenith** [comment]
Hello @devalex88 

The issue does there in 6.3.3.
To help you to reproduce the issue, here are the reproduce steps.

**1. Script**
Download the script: 
[retry_script.zip](https://github.com/katalon-studio/katalon-studio/files/3619957/retry_script.zip)

**2. Run in console**
Here is the command to run in console.
Make sure retry happens, and the last try is PASS status.
```
${KATALON_KATALON_INSTALL_DIR}/katalon -statusDelay=60 -runMode=console -projectPath=/Users/ken/Desktop/TMP/retry -testSuiteCollectionPath="Test Suites/retryCollection" -apiKey=27e853f3-33f6-4153-a0ff-2f692b5a44b3 -retry=3
```

**3. Check Exits Code**
You could find the exits code is 1, even the last retry PASS.
```
ret_code=$?
echo $ret_code
# 1
```

**4. Logs***
Here is the log. At last retry the test is success, but exit code is still 1.

```
root@a336e1569a36:/Users/ken/Desktop/TMP/retry# ${KATALON_KATALON_INSTALL_DIR}/katalon -statusDelay=60 -runMode=console -projectPath=/Users/ken/Desktop/TMP/retry -testSuiteCollectionPath="Test Suites/retryCollection" -apiKey=27e853f3-33f6-4153-a0ff-2f692b5a44b3 -retry=3
Katalon workspace folder is set to default location: /opt/katalonstudio/config
Starting Groovy-Eclipse compiler resolver.  Specified compiler level: unspecified
170 2.4.7.xx-201611170128-e46 = ACTIVE


INFO: Katalon Version: 6.3.3
INFO: Command-line arguments: -statusDelay=60 -runMode=console -projectPath=/Users/ken/Desktop/TMP/retry -testSuiteCollectionPath=Test Suites/retryCollection -apiKey=****** -retry=3
INFO: User working dir: /Users/ken/Desktop/TMP/retry
INFO: User home: /root
INFO: Java vendor: Oracle Corporation
INFO: Java version: 1.8.0_212
INFO: Local OS: Linux 64bit
INFO: CPU load: 58%
INFO: Total memory: 1998 MB
INFO: Free memory: 792 MB


Start reloading plugins...
Latest plugins responses: [{"id":24703,"status":1,"expiration_date":null,"created_at":"2019-09-16T09:18:06.650Z","updated_at":"2019-09-16T09:18:06.650Z","active":true,"send_expiration_mail":false,"send_expiration_reminder_mail":false,"user_id":16058,"bundled_plugin_id":null,"product":{"id":59,"plugin_id":null,"name":"Basic Report","description":"Basic Report Plugin is a Custom Keyword that replaces for the current Report feature of Katalon Studio. Since version 6.1.5, the Report feature has been migrated to Basic Report plugin.","logo":"m3j3hddvlvinco5i-BasicReport_140px.png","repository":"https://github.com/katalon-studio/katalon-studio-basic-report-plugin","documentation":"https://github.com/katalon-studio/katalon-studio-report-plugin/blob/master/docs/tutorials/usage.md","reference":null,"license":"","release_date":null,"qa_url":null,"status":1,"created_at":"2019-04-26T09:41:28.208Z","updated_at":"2019-08-15T09:07:47.122Z","overview":"Basic Report Plugin is a Custom Keyword that replaces for the current Report feature of Katalon Studio. Since [version 6.1.5](https://docs.katalon.com/katalon-studio/new/version-615.html), the Report feature has been migrated to Basic Report plugin.  Users need to install this plugin to continue using the feature.\r\n\r\n### Basic Report Plugin provides:\r\n- Generates automatically report from Test Suite report after every test execution with various formats: HTML, CSV, JUnit, and PDF.\r\n- Allows users to export manually Test Suite and Test Suite collection report to HTML, CSV, JUnit, and PDF. Right click on the report and choose Export as.\r\n\r\nFor more details, refer to [Basic Report document](https://docs.katalon.com/katalon-studio/docs/basic-report.html).\r\n","price":null,"stripe_product_id":null,"reject_reason":null,"latest_version_number":null,"product_type_id":2,"marketplace_id":null,"product_versions":[{"id":118,"number":"1.0.2","source":"gx1r20f21bg76qli-katalon-studio-report-plugin.zip","status":1,"created_at":"2019-05-15T04:35:14.552Z","updated_at":"2019-08-15T09:11:41.634Z","min_app_version":"6.1.5","max_app_version":null,"change_log":""},{"id":99,"number":"1.0.1","source":"lazh3fjfjz5s1rq7-katalon-studio-report-plugin.jar.zip","status":1,"created_at":"2019-04-26T09:41:28.242Z","updated_at":"2019-08-15T09:11:41.637Z","min_app_version":"6.1.5","max_app_version":null,"change_log":""},{"id":145,"number":"1.0.5","source":"taschuj5zkw1sm7z-katalon-studio-report-plugin.zip","status":1,"created_at":"2019-06-20T08:44:52.820Z","updated_at":"2019-08-15T09:11:41.629Z","min_app_version":"6.2.0","max_app_version":null,"change_log":""},{"id":129,"number":"1.0.4","source":"4uc4qvqaug77t726-katalon-studio-basic-report-plugin.zip","status":1,"created_at":"2019-05-24T09:48:58.985Z","updated_at":"2019-08-15T09:11:41.630Z","min_app_version":"6.2.0","max_app_version":null,"change_log":""},{"id":127,"number":"1.0.3","source":"y5cazuxp8qzqqxo2-katalon-studio-basic-report-plugin.zip","status":1,"created_at":"2019-05-24T02:43:18.285Z","updated_at":"2019-08-15T09:11:41.632Z","min_app_version":"6.1.5","max_app_version":null,"change_log":""},{"id":186,"number":"1.0.6","source":"kzxu34vz9lggwtiq-katalon-studio-report-plugin.jar.zip","status":1,"created_at":"2019-08-15T09:07:34.663Z","updated_at":"2019-08-15T09:11:41.627Z","min_app_version":"6.3.0","max_app_version":null,"change_log":"- Fixed issue \"Cannot export PDF file without image attached\"\r\n- Fixed issue \"Cannot export PDF report if a test step's message contains any special characters\""}],"product_type":{"id":2,"name":"Custom Keywords Plugin","description":"Custom Keywords Plugin","sequence":1,"created_at":"2019-03-14T08:13:50.036Z","updated_at":"2019-03-14T08:13:50.036Z","view_type":0},"product_categories":[{"id":2,"name":"Utilities","short_name":"","description":"Utilities","order":2,"hide":false,"use_carousel":false,"created_at":"2019-06-13T08:48:28.556Z","updated_at":"2019-09-06T09:14:14.136Z","product_join_category":{"id":22,"created_at":"2019-06-13T08:47:46.501Z","updated_at":"2019-06-13T08:47:46.501Z","product_id":59,"category_id":2}}],"logoUrl":"https://katalon-store.s3.amazonaws.com/m3j3hddvlvinco5i-BasicReport_140px.png","categories":"Utilities","url":"/product/59"},"license_type":{"id":1,"name":"Free","active":true,"day":null,"month":null,"year":null,"created_at":"2019-01-30T01:47:14.080Z","updated_at":"2019-01-30T01:47:14.080Z"},"free":true,"trial":false,"paid":false,"expired":false,"canceled":false,"uninstalled":false,"finalized":false,"compatible_versions":[{"id":118,"number":"1.0.2","source":"gx1r20f21bg76qli-katalon-studio-report-plugin.zip","status":1,"created_at":"2019-05-15T04:35:14.552Z","updated_at":"2019-08-15T09:11:41.634Z","min_app_version":"6.1.5","max_app_version":null,"change_log":""},{"id":99,"number":"1.0.1","source":"lazh3fjfjz5s1rq7-katalon-studio-report-plugin.jar.zip","status":1,"created_at":"2019-04-26T09:41:28.242Z","updated_at":"2019-08-15T09:11:41.637Z","min_app_version":"6.1.5","max_app_version":null,"change_log":""},{"id":145,"number":"1.0.5","source":"taschuj5zkw1sm7z-katalon-studio-report-plugin.zip","status":1,"created_at":"2019-06-20T08:44:52.820Z","updated_at":"2019-08-15T09:11:41.629Z","min_app_version":"6.2.0","max_app_version":null,"change_log":""},{"id":129,"number":"1.0.4","source":"4uc4qvqaug77t726-katalon-studio-basic-report-plugin.zip","status":1,"created_at":"2019-05-24T09:48:58.985Z","updated_at":"2019-08-15T09:11:41.630Z","min_app_version":"6.2.0","max_app_version":null,"change_log":""},{"id":127,"number":"1.0.3","source":"y5cazuxp8qzqqxo2-katalon-studio-basic-report-plugin.zip","status":1,"created_at":"2019-05-24T02:43:18.285Z","updated_at":"2019-08-15T09:11:41.632Z","min_app_version":"6.1.5","max_app_version":null,"change_log":""},{"id":186,"number":"1.0.6","source":"kzxu34vz9lggwtiq-katalon-studio-report-plugin.jar.zip","status":1,"created_at":"2019-08-15T09:07:34.663Z","updated_at":"2019-08-15T09:11:41.627Z","min_app_version":"6.3.0","max_app_version":null,"change_log":"- Fixed issue \"Cannot export PDF file without image attached\"\r\n- Fixed issue \"Cannot export PDF report if a test step's message contains any special characters\""}],"latest_compatible_version":{"id":186,"number":"1.0.6","source":"kzxu34vz9lggwtiq-katalon-studio-report-plugin.jar.zip","status":1,"created_at":"2019-08-15T09:07:34.663Z","updated_at":"2019-08-15T09:11:41.627Z","min_app_version":"6.3.0","max_app_version":null,"change_log":"- Fixed issue \"Cannot export PDF file without image attached\"\r\n- Fixed issue \"Cannot export PDF report if a test step's message contains any special characters\""},"current_version":{"id":186,"number":"1.0.6","source":"kzxu34vz9lggwtiq-katalon-studio-report-plugin.jar.zip","status":1,"created_at":"2019-08-15T09:07:34.663Z","updated_at":"2019-08-15T09:11:41.627Z","min_app_version":"6.3.0","max_app_version":null,"change_log":"- Fixed issue \"Cannot export PDF file without image attached\"\r\n- Fixed issue \"Cannot export PDF report if a test step's message contains any special characters\""},"latest_version":{"id":186,"number":"1.0.6","source":"kzxu34vz9lggwtiq-katalon-studio-report-plugin.jar.zip","status":1,"created_at":"2019-08-15T09:07:34.663Z","updated_at":"2019-08-15T09:11:41.627Z","min_app_version":"6.3.0","max_app_version":null,"change_log":"- Fixed issue \"Cannot export PDF file without image attached\"\r\n- Fixed issue \"Cannot export PDF report if a test step's message contains any special characters\""},"created_date":"September 16th 2019, 9:18 am","download_url":"https://plugins.katalon.com/kzxu34vz9lggwtiq-katalon-studio-report-plugin.jar.zip?Expires=1568724222&Policy=eyJTdGF0ZW1lbnQiOlt7IlJlc291cmNlIjoiaHR0cHM6Ly9wbHVnaW5zLmthdGFsb24uY29tL2t6eHUzNHZ6OWxnZ3d0aXEta2F0YWxvbi1zdHVkaW8tcmVwb3J0LXBsdWdpbi5qYXIuemlwIiwiQ29uZGl0aW9uIjp7IkRhdGVMZXNzVGhhbiI6eyJBV1M6RXBvY2hUaW1lIjoxNTY4NzI0MjIyfX19XX0_&Signature=bJbPYG6qWErQV58JY6KY6DX8CnKUqw5dfF35OkUAFMIyUla3g9v1L5vu9jiJw4VuWG7L4Tu-f7uWzfmrm33yj76YKXpwO6mJrWyDwUBjYhuXFy4FS~P91s5lqOjXArCSlBEAVrpObJALCG7g8IQI1tWtoMOUO~SJMtJc2bfUEa~jEUNTM7hOtC8k-zFNbtmCfWOgyhlfQ~CaqPWpBsK89Px4rqT~qMqs8HLmJ0NqhMA~R7aAYvZhomN3iFRXcU8PNjhGUYuImfzCMkMws6tD0HpO~eOa-IOniurvOvbtU1aM4eWq5EKemFsF~Q2P-PdX9wXfKdDNQIn06H0JRb9mKg__&Key-Pair-Id=APKAIPDLSSZCBYT7R3MA"}]
Katalon version: 6.3.3
Plugin info URL: https://store.katalon.com/api/products/ks?appVersion=6.3.3
Plugin info: {
  "expired": false,
  "productId": 59,
  "name": "Basic Report",
  "id": 24703
}
Plugin ID: 24703. Plugin location: /opt/katalonstudio/configuration/plugin/install/24703/1.0.6/katalon-studio-report-plugin.jar.
Plugin "Basic Report" has been installed.
Project path is a folder.
Examine folder 'settings'.
Examine file '.DS_Store'.
Examine folder 'Drivers'.
Examine folder 'bin'.
Examine folder 'Test Listeners'.
Examine folder 'Test Cases'.
Examine folder 'Plugins'.
Examine folder 'Include'.
Examine folder 'Checkpoints'.
Examine file 'console.properties'.
Examine folder 'Test Suites'.
Examine file '.classpath'.
Examine file 'run.sh'.
Examine folder 'Libs'.
Examine folder '.settings'.
Examine folder 'Object Repository'.
Examine file '.project'.
Examine folder 'Scripts'.
Examine folder 'Profiles'.
Examine folder 'Keywords'.
Examine file 'retry.prj'.
Found project file '/Users/ken/Desktop/TMP/retry/retry.prj'.
Examine folder 'Data Files'.
Delete folder: bin
Delete folder: Libs
Opening project file: /Users/ken/Desktop/TMP/retry/retry.prj
Warning! Please run Katalon execution command outside of the project folder.
Parsing custom keywords in Plugins folder...
Generating global variables...
Parsing custom keywords...
Project 'retry' opened
chromedriver is located at default location: /opt/katalonstudio/configuration/resources/drivers/chromedriver_linux64/chromedriver. In case your browser is updated to a newer version, please use this command to update chromdriver: --config -webui.autoUpdateDrivers=true

--------------------------------------------------------------------------------
Test Suites/retryCollection - 20190917_074345............................0/1(0%)
--------------------------------------------------------------------------------
Test Suites/retries - Chrome - 20190917_074345...........................0/1(0%)
--------------------------------------------------------------------------------

Launcher status after execution process completed: Running
Start writing execution.uuid file to folder: /Users/ken/Desktop/TMP/retry/Reports/20190917_074345/retries/20190917_074345
For real-time monitoring and better reporting capabilities please integrate this project with Katalon Analytics (more details at https://docs.katalon.com/katalon-analytics/docs/integration-with-katalon-studio.html#enable-integration)


Re-run test suite: Test Suites/retries # 1.
chromedriver is located at default location: /opt/katalonstudio/configuration/resources/drivers/chromedriver_linux64/chromedriver. In case your browser is updated to a newer version, please use this command to update chromdriver: --config -webui.autoUpdateDrivers=true
Launcher status after execution process completed: Running
Start writing execution.uuid file to folder: /Users/ken/Desktop/TMP/retry/Reports/20190917_074352/retries/20190917_074352
For real-time monitoring and better reporting capabilities please integrate this project with Katalon Analytics (more details at https://docs.katalon.com/katalon-analytics/docs/integration-with-katalon-studio.html#enable-integration)


Re-run test suite: Test Suites/retries # 2.
chromedriver is located at default location: /opt/katalonstudio/configuration/resources/drivers/chromedriver_linux64/chromedriver. In case your browser is updated to a newer version, please use this command to update chromdriver: --config -webui.autoUpdateDrivers=true
Launcher status after execution process completed: Running
Start writing execution.uuid file to folder: /Users/ken/Desktop/TMP/retry/Reports/20190917_074358/retries/20190917_074358
For real-time monitoring and better reporting capabilities please integrate this project with Katalon Analytics (more details at https://docs.katalon.com/katalon-analytics/docs/integration-with-katalon-studio.html#enable-integration)


Re-run test suite: Test Suites/retries # 3.
chromedriver is located at default location: /opt/katalonstudio/configuration/resources/drivers/chromedriver_linux64/chromedriver. In case your browser is updated to a newer version, please use this command to update chromdriver: --config -webui.autoUpdateDrivers=true
Launcher status after execution process completed: Running
Start writing execution.uuid file to folder: /Users/ken/Desktop/TMP/retry/Reports/20190917_074403/retries/20190917_074403
For real-time monitoring and better reporting capabilities please integrate this project with Katalon Analytics (more details at https://docs.katalon.com/katalon-analytics/docs/integration-with-katalon-studio.html#enable-integration)

--------------------------------------------------------------------------------
Test Suites/retryCollection - 20190917_074345..........................4/4(100%)
--------------------------------------------------------------------------------
Test Suites/retries - Chrome - 20190917_074345.........................1/1(100%)
Test Suites/retries - Chrome - 20190917_074352 - Re-run 1..............1/1(100%)
Test Suites/retries - Chrome - 20190917_074358 - Re-run 2..............1/1(100%)
Test Suites/retries - Chrome - 20190917_074403 - Re-run 3..............1/1(100%)
--------------------------------------------------------------------------------

root@a336e1569a36:/Users/ken/Desktop/TMP/retry# ret_code=$?
root@a336e1569a36:/Users/ken/Desktop/TMP/retry# echo $ret_code
1
root@a336e1569a36:/Users/ken/Desktop/TMP/retry# cd Reports/
root@a336e1569a36:/Users/ken/Desktop/TMP/retry/Reports# ls
20190917_074345  20190917_074352  20190917_074358  20190917_074403
root@a336e1569a36:/Users/ken/Desktop/TMP/retry/Reports# cat 20190917_074403/retries/20190917_074403/20190917_074403.csv 
Suite/Test/Step Name,Browser,Description,Tag,Start time,End time,Duration,Status
retries,,,,2019-09-17 07:44:06,2019-09-17 07:44:07,0.821s,INCOMPLETE
,,,,,,,
Test Cases/retry,,,,2019-09-17 07:44:07,2019-09-17 07:44:07,0.445s,PASSED
"randomNumber = current().nextInt(1, 3 + 1)",,,,2019-09-17 07:44:07,2019-09-17 07:44:07,0.014s,PASSED
log = new com.kms.katalon.core.util.KeywordUtil(),,,,2019-09-17 07:44:07,2019-09-17 07:44:07,0.008s,PASSED
log.logInfo(randomNumber.toString()),,,,2019-09-17 07:44:07,2019-09-17 07:44:07,0.006s,PASSED
"verifyEqual(randomNumber, 2)",,,,2019-09-17 07:44:07,2019-09-17 07:44:07,0.099s,PASSED
root@a336e1569a36:/Users/ken/Desktop/TMP/retry/Reports# 

```

**Comment by duyluonganh** [comment]
Hi @Kenith,

We got this point. The issue will be fixed in v7.0.0 release.

Thanks for your report

**Comment by Kenith** [comment]
Hello @duyluonganh 

Really appreciate for your help.

BTW, by take this opportunity, would you please fix one more showstopper bug in v7.0.0.

**Issue Summary:**
Currently, for console run, the Katalon is mandatory to download the plugins before every running if adding apiKey. 
But normally, our test env is behind proxy/firewall. It will failed to download the plugins, such as Base Report plugin.
We even try to add proxy settings as following, still failed.
```
export http_proxt=http://127.0.0.1:3128
export https_proxt=http://127.0.0.1:3128
```

Would you please take time to fix the issue?
Or would you please provide some walk-around?

Propose solutions to fix the issue:
1. No mandatory checking & downloading the plugins any more, if Katalon detecting the plugins are already there.
    Why: If this feature enabled, we could build these plugins into our docker images
2. If proxy is set in --config command, then download the plugin behavior will go through the proxy 
3. Allow to ignore the ssl error by default when downloading the plugin

A same issue if reported here: https://github.com/katalon-studio/katalon-studio/issues/135

**Comment by devalex88** [comment]
This has been fixed in version 7.0.

**Comment by jhowellbm** [comment]
Is there a released date for the 7.0 GA release?

**Comment by jhowellbm** [comment]
Just wanted to comment and say:  LOL, this is only going to be fixed in the paid version of the product I guess?   Pretty awful customer service.
